src/server/scripts/DragonIsles/AzureVault/boss_leymor.cpp

```diff
@@ -0,0 +1,581 @@
+            {
+                case EVENT_ERRATIC_GROWTH:
+                    DoCastAOE(SPELL_ERRATIC_GROWTH_CHANNEL);
+                    _events.ScheduleEvent(EVENT_ERRATIC_GROWTH, 22s);
+                    break;
+                case EVENT_WILD_ERUPTION:
+                    DoCastAOE(SPELL_WILD_ERUPTION);
+                    _events.ScheduleEvent(EVENT_WILD_ERUPTION, 12s);
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        DoMeleeAttackIfReady();
+    }
+private:
+    EventMap _events;
+};
+
+struct npc_ley_line_sprouts : public ScriptedAI
+{
+    npc_ley_line_sprouts(Creature* creature) : ScriptedAI(creature) { }
+
+    void JustAppeared() override
+    {
+        DoCastSelf(SPELL_LEY_LINE_SPROUT_AT);
+
+        DoCastAOE(SPELL_ARCANE_POWER, true);
+    }
+
+    void JustDied(Unit* /*killer*/) override
+    {
+        if (GetDifficulty() == DIFFICULTY_MYTHIC || GetDifficulty() == DIFFICULTY_MYTHIC_KEYSTONE)
+            me->CastSpell(me->GetPosition(), SPELL_VOLATILE_SAPLING, true);
+
+        if (TempSummon* tempSummon = me->ToTempSummon())
+        {
+            if (Unit* summoner = tempSummon->GetSummonerUnit())
+            {
+                if (Aura* aura = summoner->GetAura(SPELL_ARCANE_POWER))
+                    aura->ModStackAmount(-1);
+            }
+        }
+    }
+
+    void JustSummoned(Creature* summon) override
+    {
+        InstanceScript* instance = me->GetInstanceScript();
+        if (!instance)
+            return;
+
+        Creature* leymor = instance->GetCreature(DATA_LEYMOR);
+        if (!leymor)
+            return;
+
+        if (!leymor->IsAIEnabled())
+            return;
+
+        leymor->AI()->JustSummoned(summon);
+    }
+
+    void SpellHit(WorldObject* /*caster*/, SpellInfo const* spellInfo) override
+    {
+        if (spellInfo->Id != SPELL_EXPLOSIVE_BRAND_DAMAGE && spellInfo->Id != SPELL_ERUPTING_FISSURE_SPROUT_SELECTOR)
+            return;
+
+        me->SendPlaySpellVisualKit(SPELL_VISUAL_KIT_SPROUT_DEATH, 0, 0);
+        me->KillSelf();
+    }
+};
+
+struct npc_volatile_sapling : public ScriptedAI
+{
+    npc_volatile_sapling(Creature* creature) : ScriptedAI(creature), _castedSappyBurst(false) { }
+
+    void DamageTaken(Unit* /*attacker*/, uint32& damage, DamageEffectType /*damageType*/, SpellInfo const* /*spellInfo = nullptr*/)
+    {
+        if (me->GetHealth() <= damage)
+        {
+            damage = 0;
+            me->SetHealth(1);
+            if (!_castedSappyBurst)
+            {
+                me->CastSpell(nullptr, SPELL_SAPPY_BURST, false);
+                _castedSappyBurst = true;
+            }
+        }
+    }
+
+private:
+    bool _castedSappyBurst;
+};
+
+struct boss_leymor : public BossAI
+{
+    boss_leymor(Creature* creature) : BossAI(creature, DATA_LEYMOR), _killedArcaneTender(0) { }
+
+    void JustAppeared() override
+    {
+        InstanceScript* instance = me->GetInstanceScript();
```

Comment:
All these instancescript checks are useless, because these scripts aren’t initialised if the instance doesn’t exist (you created them with RegisterAzureVaultCreatureAI)